kubespawner/spawner.py

```diff
@@ -2951,9 +2909,20 @@ def _env_keep_default(self):
 
     def _render_options_form(self, profile_list):
         self._profile_list = self._init_profile_list(profile_list)
-        profile_form_template = Environment(loader=BaseLoader).from_string(
-            self.profile_form_template
+
+        loader = FileSystemLoader(
+            self.additional_profile_form_template_paths
+            + [os.path.join(os.path.dirname(__file__), 'templates')]
         )
+
+        if self.profile_form_template != "":
+            # Admin has custom set the profile_form_template as a templated string
+            # so we use that directly
+            profile_form_template = Environment(loader=loader).from_string(
```

Comment:
This was previously `Environment(loader=BaseLoader)` and not `FileSystemLoader`.

Should it be that the FileSystemLoader should be initialized in the `else` clause and the `if` clause declares `loader=BaseLoader` like before?

Reply:
This is necessary to use the new loaders, as it allows admins to continue to use composition of the default elements we ship. For example, in https://github.com/jupyterhub/kubespawner/pull/724/files#diff-47559ea38f5c75d2661f1c86c79b031011275298348501fcec8a7d80f23f0dbb, we `{% include "style.css" %}`, and that only works because it's the `ChoiceLoader` rather than the `BaseLoader`. The `BaseLoader` is not useful when we want to use `include`, so I think this is the right thing to do.

I've moved the instantiation to be out of the conditional though, just to be clearer.